crates/stress-test/README.md

```diff
@@ -0,0 +1,36 @@
+# Shim stress test
+
+This crate provides a way to stress test the shim.
+
+## Getting started
+
+```bash
+cargo run -p stress-test -- --help
+```
+
+```
+Usage: stress-test [OPTIONS] <SHIM>
+
+Arguments:
+  <SHIM>  Path to the shim binary
+
+Options:
+  -v, --verbose                      Show the shim logs in stderr
+  -O, --container-output             Show the container output in stdout
+  -p, --parallel <PARALLEL>          Number of tasks to create and start concurrently [0 = no limit] [default: 1]
+  -S, --serial-steps <SERIAL_STEPS>  Up to what steps to run in series [default: start] [possible values: create, start, wait, delete]
+  -n, --count <COUNT>                Number of tasks to run [default: 10]
+  -t, --timeout <TIMEOUT>            Runtime timeout [0 = no timeout] [default: 2s]
+  -h, --help                         Print help
+  -V, --version                      Print version
+```
+
+Build some shim
+```bash
+make build-wasmtime
+```
+
+then stress test it
+```bash
+cargo run -p stress-test -- $PWD/target/x86_64-unknown-linux-gnu/debug/containerd-shim-wasmtime-v1
```

Comment:
I tried `--parallel=2` flag on my local but I got:

```console
cargo run -p stress-test -- --parallel=2 $PWD/target/x86_64-unknown-linux-gnu/debug/containerd-shim-wasmtime-v1
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.15s
     Running `/home/mossaka/developer/runwasi/scripts/test-runner.sh target/debug/stress-test --parallel=2 /home/mossaka/developer/runwasi/target/x86_64-unknown-linux-gnu/debug/containerd-shim-wasmtime-v1`
target/debug/stress-test --parallel=2 /home/mossaka/developer/runwasi/target/x86_64-unknown-linux-gnu/debug/containerd-shim-wasmtime-v1
Waiting for tasks to finish.
Press Ctrl-C to terminate.
 > 1 .. Error code UNKNOWN: Others("failed to connect to notify socket: notify.sock")
 > 2 .. Error code UNKNOWN: Others("failed to get cwd")
 > 3 .. Error code UNKNOWN: Others("failed to get cwd")
 > 4 .. Error code UNKNOWN: Others("failed to get cwd")
 > 5 .. Error code UNKNOWN: Others("failed to get cwd")
 > 6 .. Error code UNKNOWN: Others("failed to get cwd")
 > 7 .. Error code UNKNOWN: Others("failed to get cwd")
 > 8 .. Error code UNKNOWN: Others("failed to get cwd")
 > 9 .. [OK]
 > 10 .. Error code UNKNOWN: Others("failed to get cwd")

1 tasks succeeded
9 tasks failed
0 tasks hanged
```

Is this expected?

Reply:
Yeah, unfortunately. 
That's because of the UB.
When we fix the UB we should be able to enable concurrency.
Unfortunately even with -p=1 you might still get failures. I think only running ALL steps in series kind of guarantees we won't hit the UB, any less you can still get failures.